contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/HiveRecordReader.java

```diff
@@ -223,6 +228,24 @@ private void init() throws ExecutionSetupException {
   @Override
   public void setup(@SuppressWarnings("unused") OperatorContext context, OutputMutator output)
       throws ExecutionSetupException {
+    final ListenableFuture<Void> result = context.runCallableAs(proxyUgi,
+      new Callable<Void>() {
+        @Override
+        public Void call() throws Exception {
+          init();
+          return null;
+        }
+      });
+    try {
+      result.get();
+    } catch (InterruptedException e) {
+      result.cancel(true);
+      // Preserve evidence that the interruption occurred so that code higher up on the call stack can learn of the
+      // interruption and respond to it if it wants to.
+      Thread.currentThread().interrupt();
```

Comment:
You're right. See, I accidentally reinforced my point around the anonymous object... :)